File: src/main/java/ar/edu/utn/frba/dds/models/domain/comunidades/gradosDeConfianza/GradoDeConfianza.java
package ar.edu.utn.frba.dds.models.domain.comunidades.gradosDeConfianza;

import lombok.Getter;
import lombok.Setter;

import javax.persistence.Column;
import javax.persistence.Embeddable;

@Setter
@Getter
@Embeddable
public class GradoDeConfianza {

    @Column(name = "Tipo_de_grado")
    private TipoDeGrado nombre;

    @Column(name = "Nivel_del_grado")
    private Integer nivel;

    public void calcularNivel(double puntajeNuevo){
        if(puntajeNuevo<=5){
            this.setNivel(1);
        }
        else{
            this.setNivel(2);
        }
    }
}

File: src/main/ts/ar/edu/utn/frba/dds/domain/comunidades/gradosDeConfianza/GradoDeConfianza.ts
import {TipoDeGrado} from "./TipoDeGrado";

export class GradoDeConfianza {
    nombre: TipoDeGrado;
    nivel: number;

    calcularNivel(puntajeNuevo: number) {
        this.nivel = (puntajeNuevo <= 5) ? 1 : 2;
    }
}